### PR TITLE
gcloud ssh now uses -- to separate ssh args

### PR DESCRIPTION
**Pull request by krzyzacy**
/assign @fejta @ixdy

### CODE REVIEW

File: jenkins/agent-ctl.sh [hunk 1]
```diff
@@ -193,7 +193,7 @@ tunnel-to-master() {
   done
   sudo netstat -anp | grep :8080 && echo "8080 already used" && exit 1 \
     || echo "Tunneling to ${MASTER}..."
-  gcloud compute ssh "${MASTER}" --ssh-flag='-L8080:localhost:8080' sleep 5 &
+  gcloud compute ssh "${MASTER}" --ssh-flag='-L8080:localhost:8080' -- sleep 5 &
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: should this instead be `--command "sleep 5"`?

Comment (krzyzacy) [inline jenkins/agent-ctl.sh]: done, seems both way works

File: jenkins/agent-ctl.sh [hunk 2]
```diff
@@ -381,9 +381,9 @@ INSTANTIATE_DONE
 
 reboot-agent() {
   echo "Rebooting ${INSTANCE}..."
-  gcloud compute ssh "${INSTANCE}" sudo reboot || gcloud compute instances reset "${INSTANCE}"
+  gcloud compute ssh "${INSTANCE}" --command="sudo reboot || gcloud compute instances reset '${INSTANCE}'"
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: this should be `--command='sudo reboot' || gcloud ...` - the right-hand branch runs locally, not remotely.

File: jenkins/agent-ctl.sh [hunk 3]
```diff
@@ -381,9 +381,9 @@ INSTANTIATE_DONE
...
   sleep 30  # TODO(fejta): still but sightly less lame
-  while ! gcloud compute ssh "${INSTANCE}" uname -a < /dev/null; do
+  while ! gcloud compute ssh "${INSTANCE}" --command='uname -a < /dev/null'; do
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: same here, I think - `< /dev/null` is applied locally, not remotely.

File: jenkins/agent-ctl.sh [hunk 4]
```diff
@@ -284,7 +284,7 @@ create-agent() {
     --machine-type="${MACHINE_TYPE}" \
     --scopes="${SCOPES}" \
     --tags='do-not-delete,jenkins'
-  while ! gcloud compute ssh "${INSTANCE}" uname -a < /dev/null; do
+  while ! gcloud compute ssh "${INSTANCE}" --command='uname -a < /dev/null'; do
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: also here

File: jenkins/agent-ctl.sh (hunk 2 above)
Comment (krzyzacy) [inline jenkins/agent-ctl.sh]: true. fixed

File: jenkins/agent-ctl.sh (hunk 3 above)
Comment (krzyzacy) [inline jenkins/agent-ctl.sh]: done.

File: jenkins/agent-ctl.sh (hunk 4 above)
Comment (krzyzacy) [inline jenkins/agent-ctl.sh]: done.

**Pull request merged**